src/network/networkd-link.c

```diff
@@ -2618,6 +2618,21 @@ static int link_set_ipv6_mtu(Link *link) {
         return 0;
 }
 
+static int link_is_static_address_configured(Link *link, Address *address) {
+        assert(link);
+        assert(address);
+
+        if (link->network) {
+                Address *net_address;
+                LIST_FOREACH(addresses, net_address, link->network->static_addresses) {
+                        if (address_equal(net_address, address))
+                                return 0;
+                }
+        }
+
+        return -ENOENT;
+}
```

Comment:
i think this should really return "bool". It answers a yes/no question, and can#t really fail...